### ISSUE TITLE
Allow user to delete their profile

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-frontend#4279]
We need API/saga/reducer code to allow a user to delete their profile.

Note: I think this button might want to be red or something more dangerous-looking. Not sure.

<img width="513" alt="screenshot 2018-01-31 14 36 44" src="https://user-images.githubusercontent.com/90871/35628602-51f449c8-0694-11e8-848b-6142c7dc7a28.png">

**Comment by willdurand** [comment]
@pwalm quick & dirty:

<img width="1552" alt="screen shot 2018-05-29 at 18 41 46" src="https://user-images.githubusercontent.com/217628/40675548-fa92f31c-636f-11e8-9301-05a2818a793f.png">

Any comments? (Note: I don't really like this modal, the overlay blue is weird...)

**Comment by bobsilverberg** [comment]
I agree that that looks really weird, and as if some malware has taken over the user's computer!

**Comment by willdurand** [comment]
I revisited the modal with better UI and content (which can still be reworked later):

![2018-05-30 11 53 44](https://user-images.githubusercontent.com/217628/40716259-3a5b2a6a-6400-11e8-85ee-6cb140b5520f.gif)
